Cannot find module '@subql/node-core/admin'

### Prerequisites

* [x] Are you running the latest version(s)? `@subql/cli/4.12.0 darwin-arm64 node-v18.19.0`
* [x] Have you searched the [documentation for your issue](https://academy.subquery.network/)?
* [x] Did you search for an existing issue or pull request?

### Description

Running a new Polkadot SubQl project from starters repo - i'm using bittensor-starter

`@subql/cli/4.12.0 darwin-arm64 node-v18.19.0`

`yarn && yarn dev`

### Details

Error during startup in Docker

```
subquery-node-1   | 2024-06-05T11:29:32.081Z <worker #0> ERROR Uncaught Exception Error: Cannot find module '@subql/node-core/admin'
subquery-node-1   | Require stack:
subquery-node-1   | - /dist/admin/admin.module.js
subquery-node-1   | - /dist/app.module.js
subquery-node-1   | - /dist/init.js
subquery-node-1   | - /dist/main.js
subquery-node-1   | - /bin/run
subquery-node-1   | /node_modules/@subql/node-core/dist/indexer/worker/worker.js:133
subquery-node-1   |     throw e;
subquery-node-1   |     ^
subquery-node-1   | 
subquery-node-1   | Error: Cannot find module '@subql/node-core/admin'
subquery-node-1   | Require stack:
subquery-node-1   | - /dist/admin/admin.module.js
subquery-node-1   | - /dist/app.module.js
subquery-node-1   | - /dist/init.js
subquery-node-1   | - /dist/main.js
subquery-node-1   | - /bin/run
subquery-node-1   |     at Module._resolveFilename (node:internal/modules/cjs/loader:1140:15)
subquery-node-1   |     at Module._load (node:internal/modules/cjs/loader:981:27)
subquery-node-1   |     at Module.require (node:internal/modules/cjs/loader:1231:19)
subquery-node-1   |     at require (node:internal/modules/helpers:177:18)
subquery-node-1   |     at Object.<anonymous> (/dist/admin/admin.module.js:13:17)
subquery-node-1   |     at Module._compile (node:internal/modules/cjs/loader:1364:14)
subquery-node-1   |     at Module._extensions..js (node:internal/modules/cjs/loader:1422:10)
subquery-node-1   |     at Module.load (node:internal/modules/cjs/loader:1203:32)
subquery-node-1   |     at Module._load (node:internal/modules/cjs/loader:1019:12)
subquery-node-1   |     at Module.require (node:internal/modules/cjs/loader:1231:19) {
subquery-node-1   |   code: 'MODULE_NOT_FOUND',
subquery-node-1   |   requireStack: [
subquery-node-1   |     '/dist/admin/admin.module.js',
subquery-node-1   |     '/dist/app.module.js',
subquery-node-1   |     '/dist/init.js',
subquery-node-1   |     '/dist/main.js',
subquery-node-1   |     '/bin/run'
subquery-node-1   |   ]
subquery-node-1   | }
subquery-node-1   | 
subquery-node-1   | Node.js v18.20.3
subquery-node-1 exited with code 7
```

--- a/packages/node-core/CHANGELOG.md
+++ b/packages/node-core/CHANGELOG.md
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+- Fix various issue in monitor service, file naming issue and export admin service from node-core
+
 ## [10.4.0] - 2024-06-05
 ### Added
 - Add monitor service to record block indexing actions in order to improve POI accuracy, and provide debug info for Admin api

--- a/packages/node-core/src/index.ts
+++ b/packages/node-core/src/index.ts
@@ -16,3 +16,4 @@ export * from './utils';
 export * from './indexer';
 export * from './subcommands';
 export * from './yargs';
+export * from './admin';

--- a/packages/node-core/src/indexer/monitor.service.test.ts
+++ b/packages/node-core/src/indexer/monitor.service.test.ts
@@ -2,6 +2,7 @@
 // SPDX-License-Identifier: GPL-3.0
 
 import fs from 'fs';
+import path from 'path';
 import {makeTempDir} from '@subql/common';
 import {NodeConfig} from '../configure';
 import {MonitorService} from './monitor.service';
@@ -53,7 +54,7 @@ describe('Monitor service', () => {
   let nodeConfig: NodeConfig;
   beforeEach(async () => {
     monitorDir = await makeTempDir();
-    nodeConfig = {monitorOutDir: monitorDir, monitorFileSize: 1024} as NodeConfig;
+    nodeConfig = {monitorOutDir: monitorDir, monitorFileSize: 1024, dbSchema: 'test-db'} as NodeConfig;
     monitorService1 = new testMonitorService(nodeConfig);
     monitorService1.resetAll();
   });
@@ -207,5 +208,21 @@ describe('Monitor service', () => {
     monitorService2.resetService();
     monitorService2.testInit();
     expect(spyResetAll).toHaveBeenCalledTimes(1);
+  });
+
+  it('should generate correct file path with sanitized dbSchema', () => {
+    const expectedPath1 = path.join(monitorDir, 'test-db-fileA.txt');
+    expect((monitorService1 as any).getFilePath('A')).toBe(expectedPath1);
+
+    const nodeConfig2 = {
+      monitorOutDir: monitorDir,
+      monitorFileSize: 1024,
+      dbSchema: 'd-1234/subquery/subquery-mainnet',
+    } as NodeConfig;
+    const monitorService2 = new testMonitorService(nodeConfig2);
+    const expectedPath2 = path.join(monitorDir, 'd-1234-subquery-subquery-mainnet-fileA.txt');
+    const indexPath = path.join(monitorDir, 'd-1234-subquery-subquery-mainnet-index.csv');
+    expect((monitorService2 as any).getFilePath('A')).toBe(expectedPath2);
+    expect((monitorService2 as any).indexPath).toBe(indexPath);
   });
 });

--- a/packages/node-core/src/indexer/monitor.service.ts
+++ b/packages/node-core/src/indexer/monitor.service.ts
@@ -40,6 +40,10 @@ export interface MonitorServiceInterface {
   createBlockStart(blockHeight: number): void;
 }
 
+function sanitizeDbName(dbName: string): string {
+  return dbName.replace(/[\\/]/g, '-');
+}
+
 @Injectable()
 export class MonitorService implements MonitorServiceInterface {
   private readonly outputPath: string;
@@ -52,7 +56,7 @@ export class MonitorService implements MonitorServiceInterface {
   constructor(protected config: NodeConfig) {
     this.outputPath = config.monitorOutDir;
     this.monitorFileSize = config.monitorFileSize * UNIT_MB;
-    this.indexPath = path.join(this.outputPath, `${this.config.dbSchema}-index.csv`);
+    this.indexPath = path.join(this.outputPath, `${sanitizeDbName(this.config.dbSchema)}-index.csv`);
     this.init();
     setMonitorService(this);
   }
@@ -383,7 +387,7 @@ export class MonitorService implements MonitorServiceInterface {
   }
 
   private getFilePath(file: keyof typeof FileLocation): string {
-    return path.join(this.outputPath, `${this.config.dbSchema}-${FileLocation[file]}`);
+    return path.join(this.outputPath, `${sanitizeDbName(this.config.dbSchema)}-${FileLocation[file]}`);
   }
 
   private updateIndex(indexEntry: IndexEntry): void {

--- a/packages/node/CHANGELOG.md
+++ b/packages/node/CHANGELOG.md
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+- Fix import admin from node-core, update dockerfile to fix monitor default directory permission issue 
+
 ## [4.6.0] - 2024-06-05
 ### Added
 - Add monitor service to record block indexing actions in order to improve POI accuracy, and provide debug info for Admin api

--- a/packages/node/Dockerfile
+++ b/packages/node/Dockerfile
@@ -30,6 +30,10 @@ RUN apk add --no-cache tini curl git && \
     yarn cache clean && \
     rm -rf /root/.npm /root/.cache
 
+# Create ./.monitor directory and set permissions
+RUN mkdir -p .monitor && \
+    chown 1000:1000 .monitor
+
 # Make the user not ROOT
 USER 1000
 

--- a/packages/node/src/admin/admin.module.ts
+++ b/packages/node/src/admin/admin.module.ts
@@ -2,7 +2,7 @@
 // SPDX-License-Identifier: GPL-3.0
 
 import { Module } from '@nestjs/common';
-import { adminControllers, adminServices } from '@subql/node-core/admin';
+import { adminControllers, adminServices } from '@subql/node-core';
 import { FetchModule } from '../indexer/fetch.module';
 
 @Module({

